### ISSUE TITLE
Diffusion Model Alignment Using Direct Preference Optimization

### ISSUE THREAD

**Issue by chufanchen** [chufanchen/read-paper-and-code#52]
https://arxiv.org/abs/2311.12908

https://github.com/SalesforceAIResearch/DiffusionDPO

**Comment by ChufanSuki** [comment]
### Background

#### Align LLM

LLMs are typically aligned to human preferences using supervised fine-tuning on demonstration data, followed by RLHF.

1. training a reward function from comparison data on model outputs to represent human preferences 
2. using reinforcement learning to align the policy mode

Previous methods:

- policy-gradient methods: expensive and require extensive hyperparameter tuning, prone to reward hacking
- sample base model answers and select based on predicted rewards to use for supervised training e.g. Sparrow
- fine-tune the policy model directly on feedback data, e.g. Alpacafarm
- fine-tune the policy model utilizing a ranking loss on preference data to directly train the policy model  e.g. dpo

The fine-tune methods match RLHF in performance.

#### Align Diffusion

- DOOL
- DRAFT and AlignProp
- RL-based: DPOK, DDPO

**Comment by ChufanSuki** [comment]
### DPO

#### Reward Modeling

No access to the latent reward model $r(c,x_0)$.
Have access to ranked pairs generated from some conditioning $x_0^\omega \succ x_0^l\vert c$

Bradley-Terry model:

$$
p_{\mathrm{BT}}\left(\boldsymbol{x}_0^w \succ \boldsymbol{x}_0^l \mid \boldsymbol{c}\right)=\sigma\left(r\left(\boldsymbol{c}, \boldsymbol{x}_0^w\right)-r\left(\boldsymbol{c}, \boldsymbol{x}_0^l\right)\right)
$$

where $\sigma$ is the sigmoid function. $r(\boldsymbol{c}, \boldsymbol{x}_0^l)$ is parameterized by neural network $\phi$.

$$
L_{\mathrm{BT}}(\phi)=-\mathbb{E}_{\boldsymbol{c}, \boldsymbol{x}_0^w, \boldsymbol{x}_0^l}\left[\log \sigma\left(r_\phi\left(\boldsymbol{c}, \boldsymbol{x}_0^w\right)-r_\phi\left(\boldsymbol{c}, \boldsymbol{x}_0^l\right)\right)\right]
$$
where prompt $\boldsymbol{c}$ and data pairs $\boldsymbol{x}_0^w, \boldsymbol{x}_0^l$ are from a static dataset with human-annotated labels.
